lib/socket.js

```diff
@@ -4,11 +4,12 @@ const events = require('events')
 
 const IPv4Pattern = /^((?:[0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])[.]){3}(?:[0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])$/
 
-module.exports = class UDX extends events.EventEmitter {
-  constructor () {
+module.exports = class UDXSocket extends events.EventEmitter {
+  constructor (udxHandle) {
     super()
 
-    this._handle = Buffer.allocUnsafe(binding.sizeof_udx_napi_t)
+    this._udxHandle = udxHandle
```

Comment:
Is the `Handle` postfix helpful?

Reply:
Ah, it passes the actual handle, not the JavaScript class instance. Would `_udx._handle` be better?

Reply:
wanna push that fix?